ONSdigital/ras-rm-secure-message-v2
# RAS-1316 Secure Message v2 PATCH thread

# What and why?
This Pr adds a patch end point for threads
# How to test?
Deploy this branch then map a port to your sm_v2 instance 
```kubectl port-forward <port-name> 8051:8080 ```
or run it locally. Create a thread by running the acceptance tests, or any other way you like, taking note of the ID, which you will find in the db

Then using something like postman hit the endpoint
![Screenshot 2024-10-29 at 15 41 08](https://github.com/user-attachments/assets/d70af26b-4aa0-4eea-8db3-0eb0795d373c)

Don't forget the token, for a mapped endpoint this will work

eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXJ0eV9pZCI6ImNlMTJiOTU4LTJhNWYtNDRmNC1hNmRhLTg2MWU1OTA3MGEzMSIsInJvbGUiOiJpbnRlcm5hbCJ9.iWnqIqKyWIK5tNl7x5g74lvzhERFI1L43VCpBfUOxtU

Make sure it updates and returns the thread for the payload you give, i.e
![Screenshot 2024-10-29 at 15 43 05](https://github.com/user-attachments/assets/4b4a3a1f-d362-4016-a186-53d2402bc35d)



# Jira
